bso/structural_design/fea.hpp

```diff
@@ -37,12 +41,14 @@ namespace bso { namespace structural_design {
 		void clearResponse();
 		
 		void solve(std::string solver = "SimplicialLDLT");
+		Eigen::MatrixXd solveAdjoint(Eigen::MatrixXd& ae);
 		
 		Eigen::VectorXd getDisplacements(element::load_case lc) const;
 		const std::vector<element::node*>& getNodes() const {return mNodes;}
 		std::vector<element::node*>& getNodes() {return mNodes;}
 		const std::vector<element::element*>& getElements() const {return mElements;}
 		std::vector<element::element*>& getElements() {return mElements;}
+		const unsigned long getDOFCount() {return mDOFCount;}
```

Comment:
```suggestion
		const unsigned long& getDOFCount() const {return mDOFCount;}
```

Reply:
this way the variable get returned by reference (the ampersand) and the second `const` signifies that this function does not change the class of which it is a member.